util/kubeconfig/kubeconfig_test.go

```diff
@@ -161,47 +160,25 @@ func TestNew(t *testing.T) {
 
 	for _, tc := range testCases {
 		caKey, err := certs.NewPrivateKey()
-		if err != nil {
-			t.Fatalf("Failed to generate private key for ca cert, %v", err)
-		}
+		g.Expect(err).NotTo(HaveOccurred())
 
 		caCert, err := getTestCACert(caKey)
-		if err != nil {
-			t.Fatalf("Failed to generate test ca cert, %v", err)
-		}
+		g.Expect(err).NotTo(HaveOccurred())
 
 		actualConfig, actualError := New(tc.cluster, tc.endpoint, caCert, caKey)
 		if tc.expectError {
-			if actualError == nil {
-				t.Fatalf("Expected error but go nil error")
-			} else {
-				continue
-			}
+			g.Expect(actualError).NotTo(HaveOccurred())
```

Comment:
I think this should be an expectation for the error to have occurred looking at the original semantics 🤔 WDYT?
```suggestion
			g.Expect(actualError).To(HaveOccurred())
```

Reply:
right!